web/src/lib/stores/asset-interaction.store.ts

```diff
@@ -67,31 +67,60 @@ function createAssetInteractionStore() {
   };
 
   const navigateAsset = async (direction: 'next' | 'previous') => {
-    let index = _assetGridState.assets.findIndex(({ id }) => id === _viewingAssetStoreState.id);
-
-    index = direction === 'next' ? index + 1 : index - 1;
-
-    const needMoreAbove = index < 0;
-    const needMoreBelow = index >= _assetGridState.assets.length;
-
-    // Try to load more assets if we're at the end.
-    if (needMoreAbove || needMoreBelow) {
-      for (const bucket of _assetGridState.buckets) {
-        if (bucket.assets.length === 0) {
-          await assetStore.getAssetsByBucket(
-            bucket.bucketDate,
-            needMoreAbove ? BucketPosition.Above : BucketPosition.Below,
-          );
-          navigateAsset(direction);
-          break;
-        }
-      }
+    const currentBucketIndex = _assetGridState.loadedAssets[_viewingAssetStoreState.id];
+    if (currentBucketIndex < 0 || currentBucketIndex >= _assetGridState.buckets.length) {
+      return;
+    }
+
+    const currentBucket = _assetGridState.buckets[currentBucketIndex];
+
+    const index = currentBucket.assets.findIndex(({ id }) => id == _viewingAssetStoreState.id);
+    if (index === -1) {
       return;
     }
 
-    const asset = _assetGridState.assets[index];
-    if (asset) {
-      setViewingAsset(asset);
+    if (direction === 'next') {
+      if (index + 1 < currentBucket.assets.length) {
+        const asset = currentBucket.assets[index + 1];
+        if (asset) {
+          setViewingAsset(asset);
+        }
```

Comment:
There are a lot of nested if statements here and a lot of similar conditions. It would be more readable if it was flattened. You could just set asset to a variable and only do setViewingAsset once at the end of the file if it is not null. You could also do something like add an internal method for getNextAsset and getPreviousAsset which could return early from if statements, which would further cut down on the nesting (no more else statements in those implementations).

Reply:
> There are a lot of nested if statements here and a lot of similar conditions. It would be more readable if it was flattened. You could just set asset to a variable and only do setViewingAsset once at the end of the file if it is not null. You could also do something like add an internal method for getNextAsset and getPreviousAsset which could return early from if statements, which would further cut down on the nesting (no more else statements in those implementations).

Thank you for the idea! I believe you meant something like I just pushed. This code is better, IMO

Reply:
Yup, exactly. Looks a lot better. Thanks!

Reply:
I was going to comment the same thing, thanks for mentioning this @jrasm91.